dist/shader.vert

```diff
@@ -1,9 +1,12 @@
 attribute vec2 aVertexPosition;
 
 varying vec2 vTextureCoord;
+uniform vec2 uScale;
+uniform vec2 uTranslation;
 
 void main(void)
 {
-    gl_Position = vec4(aVertexPosition, 0.0, 1.0);
+
+    gl_Position = vec4((aVertexPosition * uScale) + uTranslation, 0.0, 1.0);
```

Comment:
I see this is a tough one! The translation within webGL is mathematically tough!

What about... would it work to do the panning outside of the image processing flow -- like, after it's been rendered, could we pan around a really large HTML element? Would that be easier? Could we use something like this library to do that? (it uses CSS i believe?)

https://cmorillas.github.io/panzoom/

What do you think? Can you tell us a bit more about what the issues are that you're trying to solve in the panning? What specifically makes you feel it's not working well, or what do you imagine it to be if it were working well?

Reply:
Thank you @jywarren. Attributing mouse coordinates to the WebGL texture was a headache. The [https://cmorillas.github.io/panzoom/](https://cmorillas.github.io/panzoom/) is a life saver. Thank you!